### ISSUE TITLE
Add an `ignoreComments` option to `no-tabs` rules

### ISSUE THREAD

**Issue by jblandry** [eslint/eslint#11562]
**What rule do you want to change?**
no-tabs

**Does this change cause the rule to produce more or fewer warnings?**
Fewer warnings

**How will the change be implemented? (New option, new default behavior, etc.)?**
Add a `ignoreComments` option 

**Please provide some example code that this change will affect:**

```js
//if ($x === true) {
//	console.log('hey');
//}
```

**What does the rule currently do for this code?**
Fails on line 2

**What will the rule do after it's changed?**
Accept tabs in comments, useful for when indentation is in tabs and we comment chunks of code

**Are you willing to submit a pull request to implement this change?**
No

**Comment by jahorwitz** [comment]
I'm going to take a stab at this

**Comment by jblandry** [comment]
5 hours between the time I raised this issue and your PR.... 

@jahorwitz You ROCK!
